src/Core/Content/Product/Cart/ProductCartProcessor.php

```diff
@@ -221,14 +223,18 @@ private function enrich(
 
         $lineItem->setQuantityInformation($quantityInformation);
 
+        $grossPurchasePrice = null;
+        if ($lineItem->getPurchasePrice()) {
+            $grossPurchasePrice = $product->getPurchasePrices()->getCurrencyPrice($context->getCurrency()->getId())->getGross();
+        }
         $lineItem->replacePayload([
             'isCloseout' => $product->getIsCloseout(),
             'customFields' => $product->getCustomFields(),
             'createdAt' => $product->getCreatedAt()->format(Defaults::STORAGE_DATE_TIME_FORMAT),
             'releaseDate' => $product->getReleaseDate() ? $product->getReleaseDate()->format(Defaults::STORAGE_DATE_TIME_FORMAT) : null,
             'isNew' => $product->isNew(),
             'markAsTopseller' => $product->getMarkAsTopseller(),
-            'purchasePrice' => $product->getPurchasePrice(),
+            'purchasePrice' => $grossPurchasePrice,
```

Comment:
I would not save the purchase price on the line item, because it is not relevant for the calculation and makes the line item itself unnecessarily large.

The purchase price should only be saved in the payload. In the payload you can save the gross and net price of the current currency price.